### PR TITLE
[TEO-183] CRUD AccessCodes

### PR DESCRIPTION
**Pull request by radulescuandrew**
https://wearetribus.atlassian.net/browse/TEO-183

### CODE REVIEW

File: backend/src/api/organization/access-code.controller.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  @Get('access-code')
+  async getAll(): Promise<AccessCodePresenter[]> {
+    const accessCodes = await this.findAllAccessCodeUseCase.execute({
+      organizationId: '3631315f-02f1-42c9-a418-8bff2e15fb2d', // TODO: replace with organization from @User request
+    });
+
+    return accessCodes.map((accessCode) => new AccessCodePresenter(accessCode));
+  }
+
+  @ApiParam({ name: 'accessCodeId', type: 'string' })
```
Comment (birloiflorian) [inline backend/src/api/organization/access-code.controller.ts:42]: Here we should have the validation uuid pipe

File: backend/src/api/organization/access-code.controller.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      code,
+      startDate,
+      endDate,
+      organizationId: '3631315f-02f1-42c9-a418-8bff2e15fb2d', // TODO: replace with organization from @User request
+      createdById: '6e5ca126-2c04-4403-a641-53345da26ef8', // TODO: replace with user from @User request
+    });
+    return new AccessCodePresenter(accessCodeModel);
+  }
+
+  @ApiParam({ name: 'id', type: 'string' })
```
Comment (birloiflorian) [inline backend/src/api/organization/access-code.controller.ts:68]: Here we should have a uuid validation pipe

File: backend/src/api/organization/access-code.controller.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  ): Promise<AccessCodePresenter> {
+    const accessCodeModel = await this.updateAccessCodeUseCase.execute({
+      id: accessCodeId,
+      endDate,
+    });
+    return new AccessCodePresenter(accessCodeModel);
+  }
+
+  @ApiParam({ name: 'id', type: 'string' })
+  @Delete(':id/access-code/:accessCodeId')
```
Comment (birloiflorian) [inline backend/src/api/organization/access-code.controller.ts:83]: Uuid validation pipe

File: backend/src/api/organization/dto/create-access-code.dto.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  IsDate,
+  IsOptional,
+  IsString,
+  MaxLength,
+  MinLength,
+} from 'class-validator';
+
+export class CreateAccessCodeDto {
+  @IsString()
+  @MaxLength(10)
```
Comment (birloiflorian) [inline backend/src/api/organization/dto/create-access-code.dto.ts:11]: If these are the real validations we need to change the story: https://wearetribus.atlassian.net/browse/TEO-36

**Review by birloiflorian — CHANGES_REQUESTED** [review]

File: backend/src/api/organization/dto/create-access-code.dto.ts (hunk 4 above)
Comment (radulescuandrew) [reply backend/src/api/organization/dto/create-access-code.dto.ts:11]: Updated the story.

File: backend/src/app.module.ts [hunk 5]
```diff
@@ -33,6 +34,7 @@ import {
     // Modules
     MailModule,
     AuthenticationModule,
+    UserModule,
```
[the remark this answers is not in the document]
Comment (radulescuandrew) [reply backend/src/app.module.ts:37]: Relation AccessCodeEntity => AdminUserEntity. Sa nu importam modulul de User in Organization

`adminUser: AdminUserEntity;`

File: backend/src/api/organization/access-code.controller.ts (hunk 1 above)
Comment (radulescuandrew) [reply backend/src/api/organization/access-code.controller.ts:42]: Am adaugat inclusiv in OrganizationStructure (BDR), o sa fie in celalalt PR.

**Review by birloiflorian — APPROVED** [review]

**Pull request merged**